[DOCS] Case enhancements in 8.3 (#2050)
elastic/security-docs: docs/cases/cases-manage.asciidoc

[[cases-open-manage]]
= Open and manage cases

[float]
[[cases-ui-open]]
== Open a new case

Open a new case to keep track of security issues and share their details with
colleagues.

. Go to *Investigate* -> *Cases*, then click *Create case*. If no cases exist, the Cases table will be empty and you'll be prompted to create one by clicking the *Create case* button inside the table.
. Give the case a name, add relevant tags, assign a severity, and provide a description.
+
TIP: In the `Description` area, you can use
https://www.markdownguide.org/cheat-sheet[Markdown] syntax.

. Choose whether you want alert statuses to sync with the case's status after they are added to the case. This option is enabled by default, but you can still turn it off after creating the case.

> . From *External incident management*, select a <<cases-external-connectors,connector>>. If you’ve previously added one, that connector displays as the default selection. Otherwise, the default setting is `No connector selected`.

Author: ```suggestion
. From *External incident management*, select a <<cases-ui-integrations, connector>>. If you’ve previously added one, that connector displays as the default selection. Otherwise, the default setting is `No connector selected`.
```
```suggestion
. From *External incident management*, select a <<cases-external-connectors,connector>>. If you’ve previously added one, that connector displays as the default selection. Otherwise, the default setting is `No connector selected`.
```

Author: Resolving because duplicate edit.

. Click *Create case*.

[role="screenshot"]
image::images/cases-ui-open.png[Shows an open case]

[float]
[[cases-ui-manage]]
== Manage existing cases

From the Cases page, you can search existing cases and filter them by tags, reporter, and status:
*open*, *in-progress*, or *closed*.

To explore a case, click on its name. You can then:

* <<cases-summary>>
* Add a new comment
* <<cases-lens-visualization>>
* Edit existing comments and the case's description

> * Add a <<cases-external-connectors,connector>> (if you did not select one while creating the case)

Author: ```suggestion
* Add a <<cases-ui-integrations, connector>> (if you did not select one while creating the case)
```
```suggestion
* Add a <<cases-external-connectors,connector>> (if you did not select one while creating the case)
```

> [[cases-open-manage]]

Reviewer: For some reason, the mini-nav on this page only includes H2 levels, but in the current live docs they also have H3s. This seems worth troubleshooting, because there are a lot of action-based sub-sections here that would be helpful to see in the mini-nav.

PR's preview page (only includes H2s):
<img width="322" alt="image" src="https://user-images.githubusercontent.com/87339667/173621268-a77f9d08-ef25-48a0-b7d7-9a200f43eef8.png">

Current live page (includes H2s and H3s):
<img width="260" alt="image" src="https://user-images.githubusercontent.com/87339667/173621337-42aee278-2a2e-4c29-8ddb-6e3255c375bc.png">

Author: Yeah, I was looking into this yesterday but didn't want to delay the review any further so I published it as is. I'm still not sure why the H3s aren't showing up so I'm going to need to poke around a bit more on this. If anyone has faced this and fixed it before, please share your tips!

Reviewer: I believe it's related to how the various index pages, `includes`, and `leveloffset` settings are interacting. It'll take some tinkering and maybe re-architecting to get those sub-topics to appear in the mini-nav, but I see it as somewhat lower priority "nice to have" at this point, and definitely not something to delay the rest of the changes in this PR.

Author: Thanks for investigating this @joepeeples ! Your notes match the observations I've made and I'm in full agreement for re-architecting the index files after the release. I still haven't figured out how the cases index file is interacting with the main and mini TOC and it'll take a lot more tweaking and testing than we have time for atm. After this release, I think it'd be worth it to consolidate the cases and Timeline index files so we can enforce the same hierarchy structure across the topics.

= Open and manage cases

You can create and manage cases using the UI or the <<cases-api-overview>>.

[float]
[[cases-ui-open]]
== Open a new case

Open a new case to keep track of security issues and share their details with
colleagues.

> . Go to *Investigate* -> *Cases*, then click *Create case*. If no cases exist, the Cases table will be empty and you'll be prompted to create one by clicking the *Create case* button inside the table.

Reviewer 2: ```suggestion
. Go to *Investigate* -> *Cases*, then click *Create case*. If no cases exist, the Cases table will be empty and you'll be prompted to create one by clicking *Create case* inside the table.
```

> . Give the case a name, add relevant tags, assign a severity level, and provide a description.
> +
> NOTE: If you do not assign your case a severity level, it will be assigned *Low* by default.
> 
> +
> TIP: In the case description, you can use
> https://www.markdownguide.org/cheat-sheet[Markdown] syntax and insert a Timeline link (click the image:images/add-timeline-button.png[Click Timeline icon,17,17] button).

Reviewer: Using Markdown and inserting a Timeline link are two different things, but the Tip mashes them together. I think the Markdown part is more essential info about how the description field works, so could be part of the step above, whereas inserting a link seems more like a Tip. Also, per Ponderisms guidelines, "If there is no text next to an actionable, visual symbol, it's an icon."
```suggestion
. Give the case a name, add relevant tags, assign a severity level, and provide a description. You can use
https://www.markdownguide.org/cheat-sheet[Markdown] syntax in the case description.
+
NOTE: If you do not assign your case a severity level, it will be assigned *Low* by default.

+
TIP: You can insert a Timeline link in the case description by clicking the Timeline icon (image:images/add-timeline-button.png[Timeline icon,17,17]).
```

. Choose whether you want alert statuses to sync with the case's status after they are added to the case. This option is enabled by default, but you can still turn it off after creating the case.

> . From *External incident management*, select a <<cases-ui-integrations, connector>>. If you’ve previously added one, that connector displays as the default selection. Otherwise, the default setting is `No connector selected`.

Author: Good point. IIRC, if you select a connector while creating a case, the "push" automatically happens once you create the case.

. Click *Create case*.

[role="screenshot"]
image::images/cases-ui-open.png[Shows an open case]

[float]
[[cases-ui-manage]]
== Manage existing cases

> From the Cases page, you can search existing cases and filter them by severity, tags, reporter, and status: *open*, *in-progress*, or *closed*. General case metrics, including how long it takes to close cases, are provided above the table.

Reviewer: To match UI. Also I don't think bold is needed here bc we're not telling the user to interact with them:
```suggestion
From the Cases page, you can search existing cases and filter them by severity, tags, reporter, and status: Open, In progress, or Closed. General case metrics, including how long it takes to close cases, are provided above the table.
```

[role="screenshot"]
image::images/cases-home-page.png[Case UI Home]

To explore a case, click on its name. You can then:

* <<cases-summary>>
* Add and <<cases-manage-comments, manage comments>>
+

> TIP: Comments can contain Markdown sytax. For syntax assistance, click the Markdown button (image:images/markdown-icon.png[Click markdown icon,17,17]) in the bottom right corner of the comment area.

Reviewer: Typo, etc.:
```suggestion
TIP: Comments can contain Markdown. For syntax help, click the Markdown icon (image:images/markdown-icon.png[Click markdown icon,17,17]) in the bottom right of the comment.
```

* <<cases-examine-alerts, Examine alerts attached to the case>>
* <<cases-lens-visualization>>
* Modify the case's description

> * Add a <<cases-ui-integrations, connector>> (if you did not select one while creating the case)

Reviewer 3: You can also select another connector even if you selected one when creating the case.

Author: Ah yes, thanks for the reminder! This is one action I always forget users can do. Thanks @cnasikas !

Author: Updated this line to "Manage connectors" and added a brief section to the "Configure external connections" topic for adding new connectors.

* Send updates to external systems (if external connections are configured)
* Close the case or reopen it
* Edit tags
* Update the case's severity
* Refresh the case to retrieve the latest updates

[float]
[[cases-summary]]
=== Review the case summary

> Click on an existing case to access the case summary. The case summary, located under the case title, contains metrics that summarize alert information and response times. These metrics update when you attach additional unique alerts to the case, add more connectors, or modify the case's status:

Reviewer 2: ```suggestion
Click on an existing case to access its summary. The case summary, located under the case title, contains metrics that summarize alert information and response times. These metrics update when you attach additional unique alerts to the case, add connectors, or modify the case's status:
```
Not sure what "unique" means here, does it mean an alert from a rule that isn't referenced in the case yet?

Author: @benironside that's sorta the idea. Each alert has a unique ID and the total number of these unique IDs is what makes up the **Total alerts** metric. For example, a user could add the same alert (Alert A) to a case multiple times. However, because they're adding the same alert, only one alert ID is counted and the **Total alert** count for the case would be 1.

> * **Total alerts**: Total number of unique alerts attached to the case.

Reviewer 2: Maybe I'm missing something but I'm wondering why this list has periods after each item when the list in the previous section does not.

Author: This is an oversight on my part. I can fix it, thanks!

* **Associated users**: Total number of unique users that are represented in the attached alerts.
* **Associated hosts**: Total number of unique hosts that are represented in the attached alerts.
* **Total connectors**: Total number of connectors that have been added to the case.
* **Case created**: Date and time that the case was created.
* **Open duration**: Time elapsed since the case was created.
* **In progress duration**: How long the case has been in the `In progress` state.
* **Duration from creation to close**: Time elapsed from when the case was created to when it was closed.

[role="screenshot"]
image::images/cases-summary.png[Shows you a summary of the case]

[float]
[[cases-manage-comments]]
=== Manage case comments

> To edit, delete, or quote a comment, click the *More actions* menu (*…​*) and select the appropriate option.

Reviewer 2: ```suggestion
To edit, delete, or quote a comment, click the *More actions* menu (*…​*).
```

Reviewer 4: Alternate suggestion here: 
```suggestion
To edit, delete, or quote a comment, select the appropriate option from the *More actions* menu (*…​*).
```

[role="screenshot"]
image::images/cases-manage-comments.png[Shows you a summary of the case]

[float]
[[cases-examine-alerts]]
=== Examine alerts attached to a case

preview::[]

To explore the alerts attached to a case, click the *Alerts* tab. In the table, alerts are organized from oldest to newest. To <<view-alert-details, view alert details>>, click the *View details* button.

[role="screenshot"]

> image::images/cases-alert-tab.gif[Shows you the Alerts tab]

Author: Yeah, I see your point. I was going back and forth on the gif as well because the alert details flyout will likely change in a future release. Omitting it here and keeping the focus on the table makes sense.

> image::images/cases-alert-tab.gif[Shows you the Alerts tab]

Author: Replaced with a static image of the **Alerts** tab.

[float]
[[cases-lens-visualization]]
=== Add a Lens visualization

beta[]

> Add a Lens visualization to your case to portray event and alert data through charts and graphs.

Reviewer 2: ```suggestion
Add a Lens visualization to your case to visualize event and alert data using charts and graphs.
```

[role="screenshot"]

> image::images/add-vis-to-case.gif[Shows how to add a visualization to a case]

Author: Saving this change for last because it takes a few tries to nail the gif.

To add a Lens visualization to a comment within your case:

. Click the *Visualization* button. The Add visualization dialog appears. 
. Select an existing visualization from your Visualize Library or create a new visualization.

+

> IMPORTANT: Set an absolute time range for your visualization. This ensures your visualization doesn't change over time after you save it to your case, and provides important context for others managing the case.

Reviewer 2: ```suggestion
IMPORTANT: Set absolute time ranges for your visualizations. This ensures they won't change after being saved to your case, and provides important context for others managing the case.
```

+
. Save the visualization to your Visualize Library by clicking the *Save to library* button (optional).
.. Enter a title and description for the visualization. 

> .. Choose whether you want to keep the *Update panel on Security* activated. This option is activated by default and automatically adds the visualization to your Visualize Library.

Reviewer 2: ```suggestion
.. Choose whether to keep *Update panel on Security* activated. This option is active by default and automatically adds the visualization to your Visualize Library.
```

. After you've finished creating your visualization, click *Save and return* to go back to your case.
. Click *Preview* to show how the visualization will appear in the case comment.
. Click *Add Comment* to add the visualization to your case. 

NOTE: Once a visualization has been added to a case, it cannot be modified or deleted. However, you can interact with the visualization by clicking the *Open Visualization* option in the comment menu.
  
[role="screenshot"]
image::images/cases-open-vis.png[Shows where the Open Visualization option is]

[float]
[[cases-export-import]]
== Export and import cases

> Cases can be <<cases-export, exported>> and <<cases-import, imported>> as saved objects through the Kibana {kibana-ref}/managing-saved-objects.html[Saved Objects] UI.

Reviewer 2: ```suggestion
Cases can be <<cases-export, exported>> and <<cases-import, imported>> as saved objects via the Kibana {kibana-ref}/managing-saved-objects.html[Saved Objects] UI.
```

Reviewer 4: @nastasha-solomon yes - use `using` instead. Will update @benironside's comment.

IMPORTANT: Before importing Lens visualizations, Timelines, or alerts into a space, ensure their data is present. Without it, they won't work after being imported.

[float]
[[cases-export]]
=== Export a case
Use the *Export* option to move cases between different Kibana instances. When you export a case, the following data is exported to a newline-delimited JSON (`.ndjson`) file: case details, user actions, text string comments, case alerts, and lens visualizations (which are exported as JSON blobs).

To export a case:

. Open the main menu, click *Stack Management -> Kibana*, then select the *Saved Objects* tab.
. Search for the case by choosing a saved object type or entering the case title in the search bar.
. Select one or more cases, then click the *Export* button.

> . Click *Export*. A confirmation message that your file is downloading displays.

Reviewer 2: ```suggestion
. A confirmation message that your file is downloading appears.
```

+
TIP: Keep the *Include related objects* option enabled to ensure connectors are exported too.

[role="screenshot"]
image::images/cases-export-button.png[Shows the export saved objects workflow]

[float]
[[cases-import]]
=== Import a case

To import a case:

. Open the main menu, click *Stack Management -> Kibana* and then select the *Saved Objects* tab.
. Click *Import*.
. Select the NDJSON file containing the exported case and configure the import options.
. Click *Import*.
. Review the import log and click *Done*.
+
[IMPORTANT]
=========================

Be mindful of the following:

* If the imported case had connectors attached to it, you'll be prompted to re-authenticate the connectors. To do so, click *Go to connectors* on the *Import saved objects* flyout and complete the necessary steps. Alternatively, open the main menu, then go to *Stack Management -> Alerts and Insights -> Rules and Connectors -> Connectors* to access connectors.

> * If the imported case had attached alerts, verify that the alerts’ source documents are present in the environment. Case features that interact with alerts (such as the Alert Details flyout and rule details page) rely on the alerts’ source documents to function.

Reviewer 2: Not sure why Alert Details is capitalized when rule details is not

Author: The "d" in "Alert details flyout" should actually be lowercase. A while back, the alert flyout was named "Alert details"  so capitlizing the "d" is a convention from those days. I'm open to revisiting this guideline that at any time though, especially since it's been a few releases since that change. The term "rule details" has always been lowercase as far as I remember. I've noticed some inconsistencies in our other docs, so that's something we might want to address at a later time.

[[cases-open-manage]]
= Open and manage cases

You can create and manage cases using the UI or the <<cases-api-overview>>.

[float]
[[cases-ui-open]]
== Open a new case

Open a new case to keep track of security issues and share their details with
colleagues.

. Go to *Investigate* -> *Cases*, then click *Create case*. If no cases exist, the Cases table will be empty and you'll be prompted to create one by clicking the *Create case* button inside the table.
. Give the case a name, add relevant tags, assign a severity level, and provide a description. You can use
https://www.markdownguide.org/cheat-sheet[Markdown] syntax in the case description.
+
NOTE: If you do not assign your case a severity level, it will be assigned *Low* by default.

+
TIP: You can insert a Timeline link in the case description by clicking the Timeline icon (image:images/add-timeline-button.png[Timeline icon,17,17]).

> . Choose whether you want alert statuses to sync with the case's status after they are added to the case. This option is enabled by default, but you can still turn it off after creating the case.

Reviewer 4: "whether" usually indicates there are two choices you're choosing from. 

```suggestion
. Choose if you want alert statuses to sync with the case's status after they are added to the case. This option is enabled by default, but you can turn it off after creating the case.
```

. From *External incident management*, select a <<cases-ui-integrations, connector>>. If you’ve previously added one, that connector displays as the default selection. Otherwise, the default setting is `No connector selected`.
. Click *Create case*.
+
NOTE: If you've selected a connector for the case, the case is automatically pushed to the third-party system it's connected to.

[role="screenshot"]
image::images/cases-ui-open.png[Shows an open case]

[float]
[[cases-ui-manage]]
== Manage existing cases

> From the Cases page, you can search existing cases and filter them by severity, tags, reporter, and status: Open, In progress, or Closed. General case metrics, including how long it takes to close cases, are provided above the table.

Author: I think it is code text. Here's an example:
https://www.elastic.co/guide/en/security/master/admin-page-ov.html#endpoints-list-ov

preview::[]

To explore the alerts attached to a case, click the *Alerts* tab. In the table, alerts are organized from oldest to newest. To <<view-alert-details, view alert details>>, click the *View details* button.

[role="screenshot"]
image::images/cases-alert-tab.png[Shows you the Alerts tab]

[float]
[[cases-lens-visualization]]
=== Add a Lens visualization

beta[]

Add a Lens visualization to your case to portray event and alert data through charts and graphs.

[role="screenshot"]
image::images/add-vis-to-case.gif[Shows how to add a visualization to a case]

To add a Lens visualization to a comment within your case:

. Click the *Visualization* button. The Add visualization dialog appears. 
. Select an existing visualization from your Visualize Library or create a new visualization.

+

IMPORTANT: Set an absolute time range for your visualization. This ensures your visualization doesn't change over time after you save it to your case, and provides important context for others managing the case.

+
. Save the visualization to your Visualize Library by clicking the *Save to library* button (optional).
.. Enter a title and description for the visualization. 

> .. Choose whether you want to keep the *Update panel on Security* activated. This option is activated by default and automatically adds the visualization to your Visualize Library.

Reviewer 4: ```suggestion
.. Choose if you want to keep the *Update panel on Security* activated. This option is activated by default and automatically adds the visualization to your Visualize Library.
```
